moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -1974,61 +2026,59 @@ double RobotState::testJointSpaceJump(const JointModelGroup* group, std::vector<
 
   double percentage = 1.0;
   // compute the average distance between the states we looked at
-  double thres = jump_threshold * (total_dist / (double)dist_vector.size());
+  double thres = jump_threshold_factor * (total_dist / (double)dist_vector.size());
   for (std::size_t i = 0; i < dist_vector.size(); ++i)
     if (dist_vector[i] > thres)
     {
       CONSOLE_BRIDGE_logDebug("Truncating Cartesian path due to detected jump in joint-space distance");
-      percentage = (double)i / (double)dist_vector.size();
-      traj.resize(i);
+      percentage = (double)(i + 1) / (double)(traj.size());
+      traj.resize(i + 1);
       break;
     }
 
   return percentage;
 }
 
-double RobotState::computeCartesianPath(const JointModelGroup* group, std::vector<RobotStatePtr>& traj,
-                                        const LinkModel* link, const EigenSTL::vector_Affine3d& waypoints,
-                                        bool global_reference_frame, double max_step, double jump_threshold,
-                                        const GroupStateValidityCallbackFn& validCallback,
-                                        const kinematics::KinematicsQueryOptions& options)
+double RobotState::testJointSpaceJump(const JointModelGroup* group, std::vector<RobotStatePtr>& traj,
+                                      double revolute_jump_threshold, double prismatic_jump_threshold)
 {
-  double percentage_solved = 0.0;
-  for (std::size_t i = 0; i < waypoints.size(); ++i)
+  bool still_valid = true;
+  const std::vector<const JointModel*>& joints = group->getActiveJointModels();
+  for (std::size_t traj_ix = 0, ix_end = traj.size() - 1; traj_ix != ix_end; ++traj_ix)
   {
-    // Don't test joint space jumps for every waypoint, test them later on the whole trajectory.
-    static const double no_joint_space_jump_test = 0.0;
-    std::vector<RobotStatePtr> waypoint_traj;
-    double wp_percentage_solved = computeCartesianPath(group, waypoint_traj, link, waypoints[i], global_reference_frame,
-                                                       max_step, no_joint_space_jump_test, validCallback, options);
-    if (fabs(wp_percentage_solved - 1.0) < std::numeric_limits<double>::epsilon())
+    for (auto& joint : joints)
     {
-      percentage_solved = (double)(i + 1) / (double)waypoints.size();
-      std::vector<RobotStatePtr>::iterator start = waypoint_traj.begin();
-      if (i > 0 && !waypoint_traj.empty())
-        std::advance(start, 1);
-      traj.insert(traj.end(), start, waypoint_traj.end());
+      if (!joint->getType() == JointModel::PRISMATIC && !joint->getType() == JointModel::REVOLUTE)
+      {
+        CONSOLE_BRIDGE_logError("Unsupported joint type %zu in JointModelGroup %s testJointSpaceJump can only support "
```

Comment:
Throwing an exception here, seems rather hard. I think it should be perfectly valid to test a trajectory involving multi-dof joints and just ignore them. Probably there should be a warning once to emphasize the fact that we ignore them.

Reply:
I think Cartesian planning with multi-dof joints is an unusual circumstance that has not yet been addressed in moveit and probably should be in the future. However allowing it to return a solution that is wrong is very misleading and perhaps even potentially dangerous. I don't see why this is perfectly valid. Please give an example of why you would allow a multi-dof joint to not be checked for jumps.

Reply:
The joint-space jump test is only supported for revolute or prismatic joints. If the trajectory involves multi-dof joints, why shouldn't it be possible to test all the remaining, normal joints?
I'm not proposing to _silently_ ignore multi-dof joints.

Reply:
I don't think that testing for joint-space jumps for Cartesian-pose-based joints is necessary at all. If we do Cartesian interpolation for those poses with a given `max_step`, they will have the given `max_step` distance anyway...